src/UniversalIdentifier.cpp

```diff
@@ -145,9 +142,30 @@ Output::identify(transaction const& tx,
         // outputs
         std::unique_ptr<subaddress_index> subaddr_idx;
 
-        hwdev.derive_subaddress_public_key(
-					txout_key.key, derivation, i, 
-					subaddress_spendkey);
+        boost::optional<crypto::view_tag> out_vt;
+        out_vt = cryptonote::get_output_view_tag(tx.vout[i]);
+
+        // since view tags were added, we check the
+        // output's view tag against the derived view tag.
+        // If they match, we continue with deriving
+        // the subaddress spend key and checking if it
+        // matches one of ours. If no view tag match,
+        // we know the output isn't ours and can skip
+        // the expensive subaddress spend key derivation.
+        bool output_can_be_mine = true;
+        if (out_vt)
+        {
+            crypto::view_tag vt;
+            crypto::derive_view_tag(derivation, i, vt);
```

Comment:
Calling attention to this.. it doesn't use `hwdev` since the core repo doesn't support it yet. If `Output::identify` is called with a non-default `hwdev` that assumes `derivation` is encrypted, the user wouldn't be able to see their outputs. It works as is because [this is defaulted](https://github.com/moneroexamples/xmregcore/blob/be50c5257897e2a0d610317056a149f07ec894f6/src/UniversalIdentifier.hpp#L39)

I've got the core repo `hwdev` implementation sitting in the wings [over here](https://github.com/j-berman/monero/commit/cfbd590fd63ff9e0c5ec68c618e2f3fdaf24d241#diff-6cdc890becef51b38bc717331e64dd95555624f1bd139024377198169ef71ed9) waiting on ledger